server/middlewares/users-middleware.js

```diff
@@ -0,0 +1,25 @@
+import { validate, errorResponse } from '../utils/helpers';
+import registerSchema from '../utils/validators';
+/**
+ *Users endpoint middlewares
+ *
+ * @export
+ * @class UsersMiddleware
+ */
+export default class UsersMiddleware {
+  /**
+      *Validates the register endpoint
+      * @returns {undefined}
+      * @static
+      * @param {*} req
+      * @param {*} res
+      * @param {*} next
+      * @memberof UsersMiddleware
+      */
+  static validateRegister(req, res, next) {
+    validate(req.body.user, registerSchema).then(() => next()).catch((err) => {
```

Comment:
This is only really pertinent to registration, so there's no particular reason for it to be a middleware of its own

Reply:
The middleware class is dedicated to validations for separation of concerns and future validation methods will be added to it. This is so that the controller methods are decluttered and perform only one function, *"interact with the database"*